src/main/java/org/opensearch/flowframework/transport/GetWorkflowTransportAction.java

```diff
@@ -0,0 +1,100 @@
+ * in the API path but will add the ability to get the workflow and not just the status in the future
+ */
+public class GetWorkflowTransportAction extends HandledTransportAction<GetWorkflowRequest, GetWorkflowResponse> {
+
+    private final Logger logger = LogManager.getLogger(GetWorkflowTransportAction.class);
+
+    private final Client client;
+    private final NamedXContentRegistry xContentRegistry;
+
+    /**
+     * Intantiates a new CreateWorkflowTransportAction
+     * @param transportService the TransportService
+     * @param actionFilters action filters
+     * @param client The client used to make the request to OS
+     * @param xContentRegistry contentRegister to parse get response
+     */
+    @Inject
+    public GetWorkflowTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(GetWorkflowAction.NAME, transportService, actionFilters, GetWorkflowRequest::new);
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, GetWorkflowRequest request, ActionListener<GetWorkflowResponse> listener) {
+        String workflowId = request.getWorkflowId();
+        User user = ParseUtils.getUserContext(client);
+        GetRequest getRequest = new GetRequest(WORKFLOW_STATE_INDEX).id(workflowId);
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                logger.debug("Completed Get Workflow Status Request, id:{}", workflowId);
+
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = ParseUtils.createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        WorkflowState workflowState = WorkflowState.parse(parser);
+                        listener.onResponse(new GetWorkflowResponse(workflowState, request.getAll()));
+                    } catch (Exception e) {
+                        logger.error("Failed to parse workflowState" + r.getId(), e);
+                        listener.onFailure(e);
+                    }
+                } else {
+                    listener.onFailure(new FlowFrameworkException("Fail to find workflow", RestStatus.NOT_FOUND));
+                }
+            }, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    listener.onFailure(new FlowFrameworkException("Fail to find workflow", RestStatus.NOT_FOUND));
+                } else {
+                    logger.error("Failed to get workflow status of  " + workflowId, e);
+                    listener.onFailure(e);
```

Comment:
We should create a new `FlowFrameworkException` object and pass that to:

```suggestion
                    listener.onFailure(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
```

Here and for the rest of the failure in the PR.

Reply:
changed